node/signer.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package node
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ava-labs/avalanchego/config/node"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/localsigner"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/rpcsigner"
+	"github.com/ava-labs/avalanchego/utils/perms"
+)
+
+// newStakingSigner returns a BLS signer based on the provided configuration.
+func newStakingSigner(
+	config any,
+) (bls.Signer, error) {
+	switch cfg := config.(type) {
+	case node.EphemeralSignerConfig:
+		signer, err := localsigner.New()
+		if err != nil {
+			return nil, fmt.Errorf("could not generate ephemeral signer: %w", err)
+		}
+
+		return signer, nil
+
+	case node.ContentKeyConfig:
+		signerKeyContent, err := base64.StdEncoding.DecodeString(cfg.SignerKeyRawContent)
+		if err != nil {
+			return nil, fmt.Errorf("unable to decode base64 content: %w", err)
+		}
+
+		signer, err := localsigner.FromBytes(signerKeyContent)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse signing key: %w", err)
+		}
+
+		return signer, nil
+
+	case node.RPCSignerConfig:
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		signer, err := rpcsigner.NewClient(ctx, cfg.StakingSignerRPC)
+		cancel()
```

Comment:
nit: could we use `defer cancel()` instead of calling `cancel()` directly? it's technically the same but it's more idiomatic in go + helps us to avoid subtle bugs in the future if this code is modified.